app_secret.go

```diff
@@ -11,8 +11,9 @@ const appCreationNature = 1
 
 func newAppId(appSecret []byte) []byte {
 	publicKey := appSecret[AppSecretSize-AppPublicKeySize : AppSecretSize]
-	author := make([]byte, 32)
-	payload := append([]byte{appCreationNature}, author...)
+
+	payload := make([]byte, 33)
+	payload[0] = appCreationNature
```

Comment:
The previous style was better imho. I'd even make it more verbose

```go
nature := []byte{appCreationNature}
author := make([]byte, 32)
payload := make([]byte, 0)
payload := append(payload, nature..., author..., publicKey...) // can't remember if this is legal
```

The idea is that what's written here should not be too hard to correlate with [the spec](https://github.com/TankerHQ/spec/blob/master/blocks_format.md#block-format)

Reply:
More over, the actual code should not work.
```go
payload := make([]byte, 33)
payload = append(payload, publicKey...)
if len(payload) != 33 {
  assert("publicKey was appended after the 32 zeros");
} 
```
We should use `make([]byte, 0, 33)` and then the same `append()` as before

Reply:
@blastrock unfortunately what you suggest is not legal and requires chained appends, like `append(payload, append(nature, append(author, publicKey...)...)...)`
What I tried to do here is avoid the unnecessary allocation of author that's clearly only a sequence of zeros.

Following my previous note, @JMounier I'm not sure I get what you mean, the previous code had a final state of:
```
[appCreationNature, 0, 0, 0... (32 zeros in total)]
```
which is exactly what this does -> allocate a 33 bytes slice and change only the first one? Correct me if I'm wrong!

Reply:
My bad. It felt wrong to pad with 32 zeros for no obvious reason. It seems to be a placeholder for an author

Reply:
I remade a change that I think better conveys the idea of the spec while avoiding the unnecessary allocation, WDYT @blastrock ?

Reply:
that's better :)